tests/test_arrow_dataset.py

```diff
@@ -2708,6 +2708,151 @@ def test_tf_dataset_conversion(self, in_memory):
+            with self._create_dummy_dataset(in_memory, tmp_dir.name, multiple_columns=True) as dset:
+                # Check behavior for different values of return_dict
+                # If return_dict is None and there's only one column, we get a single tensor
+                tf_dataset = dset.to_tf_dataset(columns="col_1", batch_size=2, num_workers=num_workers)
+                batch = next(iter(tf_dataset))
+                self.assertEqual(batch.shape.as_list(), [2])
+                self.assertEqual(batch.dtype.name, "int64")
+                # If return_dict is None and there's a label and only one feature column, we get a tuple of tensors
+                tf_dataset = dset.to_tf_dataset(
+                    columns="col_1", label_cols="col_2", batch_size=2, num_workers=num_workers
+                )
+                batch = next(iter(tf_dataset))
+                self.assertTrue(len(batch) == 2)
+                self.assertEqual(batch[0].shape.as_list(), [2])
+                self.assertEqual(batch[0].dtype.name, "int64")
+                self.assertEqual(batch[1].shape.as_list(), [2])
+                self.assertEqual(batch[1].dtype.name, "string")
+                # If return_dict is None, and there's more than one feature column, we get a tuple, where the first element is a dict
+                # and the second element is a tuple
+                tf_dataset = dset.to_tf_dataset(
+                    columns=["col_1", "col_3"], label_cols="col_2", batch_size=2, num_workers=num_workers
+                )
+                batch = next(iter(tf_dataset))
+                self.assertEqual(batch[0]["col_1"].shape.as_list(), [2])
+                self.assertEqual(batch[0]["col_1"].dtype.name, "int64")
+                self.assertEqual(batch[0]["col_3"].shape.as_list(), [2])
+                self.assertEqual(batch[0]["col_3"].dtype.name, "int64")
+                self.assertEqual(batch[1].shape.as_list(), [2])
+                self.assertEqual(batch[1].dtype.name, "string")
+                # If return_dict is None, and there's more than one label column, we get tuple, where the first element is a tuple
+                # and the second element is a dict
+                tf_dataset = dset.to_tf_dataset(
+                    columns="col_1", label_cols=["col_2", "col_3"], batch_size=2, num_workers=num_workers
+                )
+                batch = next(iter(tf_dataset))
+                self.assertEqual(batch[0].shape.as_list(), [2])
+                self.assertEqual(batch[0].dtype.name, "int64")
+                self.assertEqual(batch[1]["col_2"].shape.as_list(), [2])
+                self.assertEqual(batch[1]["col_2"].dtype.name, "string")
+                self.assertEqual(batch[1]["col_3"].shape.as_list(), [2])
+                self.assertEqual(batch[1]["col_3"].dtype.name, "int64")
+                # If return_dict is True and there's only one column, we get a single tensor
```

Comment:
I think we always need to return a dict if `return_dict=True`

Reply:
That is actually what happens, I just wrote a bad comment 😅 . I'll double check they're all OK